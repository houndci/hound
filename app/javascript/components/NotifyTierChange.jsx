import React from 'react';

import App from './NotifyTierChange/components/App';

export default class NotifyTierChange extends React.Component {
  render() {
    return(
      <App {...this.props} />
    );
  }
}

