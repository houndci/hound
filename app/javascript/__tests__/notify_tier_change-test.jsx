import NotifyTierChange from "../components/NotifyTierChange";

let repo = {id: 1, name: "Test repo"};
let plans = [
  { name: "Chihuahua", price: 49, allowance: 4 },
  { name: "Labrador", price: 99, allowance: 10 },
  { name: "Great Dane", price: 249, allowance: 30 },
];

function make_active_plan(target_name) {
  return _.map(plans, (plan) => {
    if (plan.name === target_name) {
      plan.current = true;
      return plan;
    } else {
      plan.current = false;
      return plan;
    }
  });
}

describe("NotifyTierChange", () => {
  describe("snapshots", () => {
    it("renders plans appropriately (Chihuahua -> Labrador)", () => {
      plans = make_active_plan("Chihuahua");

      const wrapper = shallow(
        <NotifyTierChange
          authenticity_token = "csrf_token"
          plans = {plans}
          repo_id = {repo.id}
          repo_name = {repo.name}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it("renders plans appropriately (Labrador -> Great Dane)", () => {
      plans = make_active_plan("Labrador");

      const wrapper = shallow(
        <NotifyTierChange
          authenticity_token = "csrf_token"
          plans = {plans}
          repo_id = {repo.id}
          repo_name = {repo.name}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
