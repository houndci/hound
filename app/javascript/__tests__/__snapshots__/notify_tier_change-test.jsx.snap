exports[`NotifyTierChange snapshots renders plans appropriately (Chihuahua -> Labrador) 1`] = `
<App
  authenticity_token="csrf_token"
  plans={
    Array [
      Object {
        "allowance": 4,
        "current": true,
        "name": "Chihuahua",
        "price": 49,
      },
      Object {
        "allowance": 10,
        "current": false,
        "name": "Labrador",
        "price": 99,
      },
      Object {
        "allowance": 30,
        "current": false,
        "name": "Great Dane",
        "price": 249,
      },
    ]
  }
  repo_id={1}
  repo_name="Test repo" />
`;

exports[`NotifyTierChange snapshots renders plans appropriately (Labrador -> Great Dane) 1`] = `
<App
  authenticity_token="csrf_token"
  plans={
    Array [
      Object {
        "allowance": 4,
        "current": false,
        "name": "Chihuahua",
        "price": 49,
      },
      Object {
        "allowance": 10,
        "current": true,
        "name": "Labrador",
        "price": 99,
      },
      Object {
        "allowance": 30,
        "current": false,
        "name": "Great Dane",
        "price": 249,
      },
    ]
  }
  repo_id={1}
  repo_name="Test repo" />
`;
